src/profiler.cpp

```diff
@@ -1075,6 +1094,32 @@ Error Profiler::start(Arguments& args, bool reset) {
     return error;
 }
 
+Error Profiler::setContextId(u64 contextId) {
+    void *value = pthread_getspecific(local_context_key);
+
+    int status = 0;
+
+    if (!value) {
+        Context *context = (Context*) malloc(sizeof(Context));
```

Comment:
Why is there still a need to use malloc?

Reply:
```pthread_setspecific``` needs a pointer to a thread-local variable. Is it safe to add a pointer to a ```jlong``` that is passed from JNI call?

Reply:
Well, a pointer is just a number, so I think that you could simply cast the `contextId` to the pointer type and use it as such. In this case though I would suggest to switch to `u32`.

Reply:
I absolutely agree there is no need for memory allocation here.

`contextId` could remain `u64`, but what to do with Java API on 32-bit systems then? Either throw an exception if contextId does not fit `uintptr_t`, or just make it `u32` to avoid confusion, as @AdrK suggests. I don't mind either solution.